<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>chessClass</class>
 <widget class="QMainWindow" name="chessClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>676</width>
    <height>437</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chai Chess</string>
  </property>
  <property name="windowIcon">
   <iconset resource="chess.qrc">
    <normaloff>:/chess/chai_symbol</normaloff>:/chess/chai_symbol</iconset>
  </property>
  <property name="iconSize">
   <size>
    <width>128</width>
    <height>128</height>
   </size>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>3</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="Chessboard" name="widgetChessboard" native="true">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>100</width>
        <height>100</height>
       </size>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <zorder>groupBoxOptions</zorder>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>3</number>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBoxOptions">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="title">
         <string>Options</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>3</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>3</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <widget class="QLabel" name="label">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Game:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBoxGame">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QStackedWidget" name="stackedWidgetOptions">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="currentIndex">
            <number>1</number>
           </property>
           <widget class="QWidget" name="pageManual"/>
           <widget class="QWidget" name="pageGreedy">
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="spacing">
              <number>3</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="label_4">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Depth:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBoxGreedyDepth">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="suffix">
                  <string> plies</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Current Score:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditCurScore">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maxLength">
           <number>15</number>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Best Score:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditBestScore">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maxLength">
           <number>15</number>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonNewGame">
          <property name="text">
           <string>New Game</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonStopGame">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextEdit" name="textEditLog">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::NoTextInteraction</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>676</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Chessboard</class>
   <extends>QWidget</extends>
   <header>chessboard.h</header>
   <container>1</container>
   <slots>
    <signal>updateLog(QString)</signal>
    <signal>moveMade()</signal>
    <slot>newGame()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="chess.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButtonNewGame</sender>
   <signal>clicked()</signal>
   <receiver>widgetChessboard</receiver>
   <slot>newGame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>560</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>261</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonNewGame</sender>
   <signal>clicked()</signal>
   <receiver>chessClass</receiver>
   <slot>newGame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>560</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>356</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonStopGame</sender>
   <signal>clicked()</signal>
   <receiver>chessClass</receiver>
   <slot>stopGame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>673</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widgetChessboard</sender>
   <signal>updateLog(QString)</signal>
   <receiver>textEditLog</receiver>
   <slot>insertHtml(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonNewGame</sender>
   <signal>clicked()</signal>
   <receiver>textEditLog</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>547</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>541</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEditLog</sender>
   <signal>textChanged()</signal>
   <receiver>chessClass</receiver>
   <slot>onLogChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>585</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>681</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxGame</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidgetOptions</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>476</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widgetChessboard</sender>
   <signal>moveMade()</signal>
   <receiver>chessClass</receiver>
   <slot>onMove()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>newGame()</slot>
  <slot>stopGame()</slot>
  <slot>appendLog(QString)</slot>
  <slot>onLogChanged()</slot>
  <slot>onMove()</slot>
 </slots>
</ui>
